server/src/main/java/org/apache/lucene/codecs/experimental/ZstdCompressionMode.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.apache.lucene.codecs.experimental;
+
+import com.github.luben.zstd.*;
+import java.io.IOException;
+import org.apache.lucene.codecs.compressing.CompressionMode;
+import org.apache.lucene.codecs.compressing.Compressor;
+import org.apache.lucene.codecs.compressing.Decompressor;
+import org.apache.lucene.store.DataInput;
+import org.apache.lucene.store.DataOutput;
+import org.apache.lucene.util.ArrayUtil;
+import org.apache.lucene.util.BytesRef;
+
+/** Zstandard Compression Mode */
+public class ZstdCompressionMode extends CompressionMode {
+
+    private static final int NUM_SUB_BLOCKS = 10;
+    private static final int DICT_SIZE_FACTOR = 6;
+
+    private final int level;
+
+    public static final int defaultLevel = 6;
+
+    /** default constructor */
+    ZstdCompressionMode() {
+        this.level = defaultLevel;
+    }
+
+    /** compression mode for a given compression level */
+    ZstdCompressionMode(int level) {
```

Comment:
better define it `public` or `protected`

Reply:
Used `protected`. Fixed.